compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -43,6 +45,48 @@ struct AsmArgs {
     pub options_spans: Vec<Span>,
 }
 
+/// A parsed list of attributes that is not attached to any item.
+/// Used to check whether `asm!` arguments are configured out.
+pub struct AsmAttrVec(pub ast::AttrVec);
+
+impl AsmAttrVec {
+    fn parse<'a>(p: &mut Parser<'a>) -> PResult<'a, Self> {
+        let mut attributes = ast::AttrVec::new();
+        loop {
+            if p.token != token::Pound {
+                break;
+            }
```

Comment:
This can be phrased as `while`, no?

Reply:
Yes, but I think this whole loop/function needs to change, to make the parsing of attributes consistent.

@nnethercote you seem to have worked on the attribute parsing a bit (and are the current reviewer here), what would be a good way to do this? e.g. this function is private, but I think it's what should be used here:

https://github.com/rust-lang/rust/blob/5fe04cbebd593c6530a21cd7bd0e8e9503b1ffe4/compiler/rustc_parse/src/parser/attr.rs#L44

It returns an `AttrWrapper` which deliberately does not give access to the inner attributes until you call `collect_tokens` (another private function). Does it make sense to make those functions public, or should the asm attribute parsing live in `rustc_parse` instead?

Reply:
after thinking about this some more, it might makes sense to, now that parsing and validation are split, to move the asm parsing code into `rustc_parse`?